Zend/zend_API.h

```diff
@@ -1991,6 +2049,29 @@ static zend_always_inline bool zend_parse_arg_array_ht(zval *arg, HashTable **de
 	return 1;
 }
 
+static zend_always_inline bool zend_parse_arg_array_ht_or_long(
+	zval *arg, HashTable **dest_ht, zend_long *dest_long, zend_bool *is_null, bool allow_null
+) {
+	if (allow_null) {
+		*is_null = 0;
+	}
+
+	if (EXPECTED(Z_TYPE_P(arg) == IS_ARRAY)) {
+		*dest_ht = Z_ARRVAL_P(arg);
+	} else if (EXPECTED(Z_TYPE_P(arg) == IS_LONG)) {
+		*dest_ht = NULL;
+		*dest_long = Z_LVAL_P(arg);
+	} else if (allow_null && EXPECTED(Z_TYPE_P(arg) == IS_NULL)) {
+		*dest_ht = NULL;
+		*is_null = 1;
```

Comment:
Should I follow the route of `zend_parse_arg_long()` et co. and initialize `dest_long` to 0?